src/actions/sequencerInbox.integration.test.ts

```diff
@@ -0,0 +1,220 @@
+    const newResult = await getMaxTimeVariation(client, {
+      sequencerInbox: l3SequencerInbox,
+    });
+    expect(newResult).toEqual(changes);
+
+    const originalValues = {
+      delayBlocks: 5_760n,
+      delaySeconds: 86_400n,
+      futureBlocks: 12n,
+      futureSeconds: 3_600n,
+    };
+    await setMaxTimeVariation(originalValues);
+    const finalResult = await getMaxTimeVariation(client, {
+      sequencerInbox: l3SequencerInbox,
+    });
+    expect(finalResult).toEqual(originalValues);
+  });
+});
+
+describe('batch poster management', () => {
+  it('isBatchPoster successfully fetches whether or an address is a batch poster', async () => {
+    const isNotBatchPosterAddress = await isBatchPoster(client, {
+      sequencerInbox: l3SequencerInbox,
+      batchPoster: zeroAddress,
+    });
+    expect(isNotBatchPosterAddress).toBeFalsy();
+    const isBatchPosterAddress = await isBatchPoster(client, {
+      sequencerInbox: l3SequencerInbox,
+      batchPoster: l3BatchPoster,
+    });
+    expect(isBatchPosterAddress).toBeTruthy();
+  });
+
+  it('successfully enable or disable an address as batch poster', async () => {
+    const randomAddress = privateKeyToAccount(generatePrivateKey()).address;
+    const isRandomAddressBatchPoster = await isBatchPoster(client, {
+      sequencerInbox: l3SequencerInbox,
+      batchPoster: randomAddress,
+    });
+    expect(isRandomAddressBatchPoster).toBeFalsy();
+
+    const enableTransactionRequest = await buildEnableBatchPoster(client, {
+      sequencerInbox: l3SequencerInbox,
+      upgradeExecutor: l3UpgradeExecutor,
+      batchPoster: randomAddress,
+      account: l3RollupOwner.address,
+    });
+    const enableTxHash = await client.sendRawTransaction({
+      serializedTransaction: await l3RollupOwner.signTransaction(enableTransactionRequest),
+    });
+    await client.waitForTransactionReceipt({ hash: enableTxHash });
+    const isRandomAddressBatchPosterAfterEnabling = await isBatchPoster(client, {
+      sequencerInbox: l3SequencerInbox,
+      batchPoster: randomAddress,
+    });
+    expect(isRandomAddressBatchPosterAfterEnabling).toBeTruthy();
+
+    const disableTransactionRequest = await buildDisableBatchPoster(client, {
+      sequencerInbox: l3SequencerInbox,
+      upgradeExecutor: l3UpgradeExecutor,
+      batchPoster: randomAddress,
+      account: l3RollupOwner.address,
+    });
+    const disableTxHash = await client.sendRawTransaction({
+      serializedTransaction: await l3RollupOwner.signTransaction(disableTransactionRequest),
+    });
+    await client.waitForTransactionReceipt({ hash: disableTxHash });
+    const isRandomAddressBatchPosterAfterDisabling = await isBatchPoster(client, {
+      sequencerInbox: l3SequencerInbox,
+      batchPoster: randomAddress,
+    });
+    expect(isRandomAddressBatchPosterAfterDisabling).toBeFalsy();
+  });
+});
+
+describe('keyset management', () => {
+  async function deployAnytrustRollupWithKeyset(keyset: Hex) {
```

Comment:
a chain can't be rollup _and_ anytrust
```suggestion
  async function deployAnyTrustChainWithKeyset(keyset: Hex) {
```